process/transaction/metaProcess.go

```diff
@@ -20,6 +21,16 @@ type metaTxProcessor struct {
 	txTypeHandler process.TxTypeHandler
 }
 
+// GetSmartContractResults returns the smart contract results
+func (txProc *metaTxProcessor) GetSmartContractResults() map[string]data.TransactionHandler {
+	return make(map[string]data.TransactionHandler)
```

Comment:
this seems wrong.

Reply:
meta sc processor creates smart contract results as well. you do not these. As in txSimulator you could have the scrForwarder and receiptForwarders and take those data from there.

Reply:
done. refactored and created structs for a more friendly API result